tests/python/test_basic_pathlike.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+import sys
+from contextlib import contextmanager
+try:
+    # python 2
+    from StringIO import StringIO
+except ImportError:
+    # python 3
+    from io import StringIO
+import numpy as np
+import xgboost as xgb
+import unittest
+import json
+import pytest
+import testing as tm
+
+# This is the select basic tests from test_basic.py with modifications for os.PathLike
+# instead of string file paths throughout.
+dpath = 'demo/data/'
+
+try:
+    from pathlib import Path
+    dpath = Path(dpath)
+except ImportError:
+    pass
+
+pytestmark = pytest.mark.skipif(**tm.no_pathlike())
```

Comment:
skip if `no_pathlike` returns true?  Are you sure it's testing the one in here instead of the one in library?

Reply:
Since 3.4 is the minimum expectation I can remove this conditional entirely. Pathlib will always be present given that min version: https://docs.python.org/3.4/library/pathlib.html